src/ansys/fluent/core/workflow_new.py

```diff
@@ -0,0 +1,1024 @@
+        """
+        task_obj = super().__getattribute__("_task_object")
+        args = task_obj.arguments
+        if item in args():
+            return getattr(args, item)
+        return getattr(task_obj, item)
+
+    def __setattr__(self, key, value):
+        """Enable attribute assignment to task arguments."""
+        task_obj = super().__getattribute__("_task_object")
+        args = task_obj.arguments
+        if hasattr(args, key):
+            setattr(args, key, value)
+        else:
+            super().__setattr__(key, value)
+
+    def __call__(self):
+        """Execute the task when called as a function."""
+        task_obj = super().__getattribute__("_task_object")
+        return task_obj.execute()
+
+    def __getitem__(self, key):
+        task_obj = super().__getattribute__("_task_object")
+        name = super().__getattribute__("_name")
+        workflow = super().__getattribute__("_workflow")
+        parent = super().__getattribute__("_parent")
+        name_1 = name
+        name_2 = re.sub(r"\s+\d+$", "", task_obj.name().strip()) + f" {key}"
+        try:
+            task_obj = getattr(workflow.task_object, name_1)[name_2]
+            if task_obj.task_type == "Compound Child":
+                temp_parent = self
+            else:
+                temp_parent = parent
+            return make_task_wrapper(task_obj, name_1, workflow, temp_parent)
+        except LookupError:
+            task_obj = getattr(workflow.task_object, name_1)[key]
+            if task_obj.task_type == "Compound Child":
+                temp_parent = self
+            else:
+                temp_parent = parent
+            try:
+                return make_task_wrapper(
+                    getattr(workflow.task_object, name_1)[key],
+                    name_1,
+                    workflow,
+                    temp_parent,
+                )
+            except LookupError as ex2:
+                raise LookupError(
+                    f"Neither '{name_2}' nor '{key}' not found in task object '{name_1}'."
+                ) from ex2
+
+    def __delitem__(self, key):
+        self[key].delete()
+
+    def _task_list(self):
+        """Gets the display names of the child tasks of a task item."""
+        task_obj = super().__getattribute__("_task_object")
+        # This is just a precaution in case this method is directly called from the task level.
+        task_list = task_obj.task_list()
+        if task_list:
+            return _convert_task_list_to_display_names(
+                super().__getattribute__("_workflow"), task_list
+            )
+        else:
+            return []
+
+    def children(self):
+        """Get ordered list of direct child tasks.
+
+        Returns
+        -------
+        List[TaskObject]
+            Ordered list of child task wrappers, or empty list if no children.
+        """
+        if not self._task_list():
+            return []
+
+        workflow = super().__getattribute__("_workflow")
```

Comment:
If we update `__getattr__()` 's implementation we should be able to dispense with all this tricky code. That way we deal with the dynamic attribute handling in one palce. There are at least two `__getattribute__()` calls here.